app/controllers/api/web/embeds_controller.rb

```diff
@@ -20,7 +20,7 @@ def show
       begin
         oembed[:html] = Sanitize.fragment(oembed[:html], Sanitize::Config::MASTODON_OEMBED)
       rescue ArgumentError
-        return not_found
+        not_found
```

Comment:
I think the `return` is important here. With your code, what will be executed if there is an `ArgumentError` is:

```ruby
not_found

render json: oembed
```

The `render json: oembed` needs to be in the `begin` block if you do want the same behaviour